src/sentry/integrations/slack/notify_action.py

```diff
@@ -30,15 +30,25 @@ def __init__(self, *args, **kwargs):
         self.fields["workspace"].choices = workspace_list
         self.fields["workspace"].widget.choices = self.fields["workspace"].choices
 
+        # XXX(meredith): When this gets set to True, it lets the RuleSerializer
+        # know to only save if and when we have the channel_id. The rule will get saved
+        # in the task (integrations/slack/tasks.py) if the channel_id is found.
+        self._pending_save = False
+
     def clean(self):
         cleaned_data = super(SlackNotifyServiceForm, self).clean()
 
         workspace = cleaned_data.get("workspace")
         channel = cleaned_data.get("channel", "")
 
-        channel_id = self.channel_transformer(workspace, channel)
+        channel_prefix, channel_id, timed_out = self.channel_transformer(workspace, channel)
         channel = strip_channel_name(channel)
 
+        if channel_id is None and timed_out:
+            cleaned_data["channel"] = channel_prefix + channel
```

Comment:
@MeredithAnya What's the reason for setting `cleaned_data["channel"]` when we timed out? All that will tell us is what prefix we timed out on, yea?

Reply:
hmm before when we were saving the rule first i needed to make sure the channel was saved with the prefix, but I think it's okay to keep because then it's consistent with how the validated_data looks

Reply:
@MeredithAnya I still don't understand what channel represents when you time out but I'll take your word that you need this code in order to work.